addons/stock/report/report_location_barcode.xml

```diff
@@ -3,33 +3,41 @@
 <data>
 
 <template id="report_generic_barcode">
-    <t t-call="web.html_container">
-        <t t-set='nRows' t-value='8'/>
-        <t t-set='nCols' t-value='3'/>
-        <div t-foreach="[docs[x:x + nRows * nCols] for x in range(0, len(docs), nRows * nCols)]" t-as="page_docs" class="page article">
-        <t t-if="title">
-          <h2 style="text-align: center; font-size: 3em"><t t-esc="title"/></h2>
-        </t>
-        <table>
-            <t t-foreach="range(nRows)" t-as="row">
-                <tr>
-                    <t t-foreach="range(nCols)" t-as="col">
-                        <t t-set="barcode_index" t-value="(row * nCols + col)"/>
-                        <t t-if="barcode_index &lt; len(page_docs)">
-                            <t t-set="o" t-value="page_docs[barcode_index]"/>
-                        </t>
-                        <t t-else="">
-                            <t t-set="o" t-value="page_docs[0]"/>
+    <t t-call="web.basic_layout">
+        <t t-call="web.html_container">
+            <t t-set="nRows" t-value="7"/>
+            <t t-set="nCols" t-value="4"/>
+            <t t-foreach="[docs[x:x + nRows * nCols] for x in range(0, len(docs), nRows * nCols)]" t-as="page_docs">
+                <div class="o_label_sheet" style="padding: 10mm 3mm;">
+                    <table class="my-0 table table-sm table-borderless">
+                        <t t-foreach="range(nRows)" t-as="row">
+                            <tr>
+                                <t t-foreach="range(nCols)" t-as="col">
+                                    <t t-set="barcode_index" t-value="(row * nCols + col)"/>
+                                    <t t-if="barcode_index &lt; len(page_docs)">
+                                        <t t-set="o" t-value="page_docs[barcode_index]"/>
+                                    </t>
+                                    <t t-else="">
+                                        <t t-set="make_invisible" t-value="True"/>
+                                    </t>
+                                    <t t-set="barcode_size" t-value="'width:47mm;height:11mm'"/>
```

Comment:
Since `barcode_size` is always the same value, you can place this `t-set` before the first `foreach` (to avoid to re-set it multiple times) or, even better, remove it and write `'width:47mm;height:11mm'` directly in the `img_style` property (since it's the only place this value is read)